packages/@aws-cdk/aws-ecs/lib/log-drivers/splunk-log-driver.ts

```diff
@@ -130,23 +137,30 @@ export class SplunkLogDriver extends LogDriver {
    * Called when the log driver is configured on a container
    */
   public bind(_scope: CoreConstruct, _containerDefinition: ContainerDefinition): LogDriverConfig {
+    const options = stringifyOptions({
+      'splunk-token': this.props.token,
+      'splunk-url': this.props.url,
+      'splunk-source': this.props.source,
+      'splunk-sourcetype': this.props.sourceType,
+      'splunk-index': this.props.index,
+      'splunk-capath': this.props.caPath,
+      'splunk-caname': this.props.caName,
+      'splunk-insecureskipverify': this.props.insecureSkipVerify,
+      'splunk-format': this.props.format,
+      'splunk-verify-connection': this.props.verifyConnection,
+      'splunk-gzip': this.props.gzip,
+      'splunk-gzip-level': this.props.gzipLevel,
+      ...renderCommonLogDriverOptions(this.props),
+    });
+
+    if (this.props.secretToken) {
+      delete options['splunk-token'];
+    }
```

Comment:
Maybe instead of deleting, what do you think we only add it when `!this.props.secretToken` since adding is cheaper than deleting?